db/main.py

```diff
@@ -0,0 +1,85 @@
+import base64
+from sqlalchemy import create_engine, Table, MetaData, Column, String
+from sqlalchemy.sql import select
+
+
+class DbRow(object):
+    def __init__(self, url, metadata=None):
+        self.url = url
+        if(metadata is not None):
+            self.rowcount = 1
+            self.metadata = str(metadata._row[0])
+        else:
+            self.rowcount = 0
+            self.metadata = metadata
+
+
+class DbUtils(object):
+
+    # engine - May not need it as a class variable
+    engine = None
+
+    # table - May not need it as a class variable
+    mydata = None
+
+    # connection - May not need it as a class variable
+    connection = None
+
+    def establishDbConnection(self):
+        # Establish connection
+        connectionString = 'postgres://' + base64.b64decode(self.username) + ':' + base64.b64decode(self.password) + '@' + base64.b64decode(self.hostName) + '/' + self.dbName
+        self.engine = create_engine(connectionString)
+        self.connection = self.engine.connect()
+
+    def __init__(self):
+        credentials = []
+        # Open credentials file. Should be stored as "username:password"
+        with open('credentials.txt') as f:
+            for line in f:
+                line = line.split("\n", 1)[0]
+                credentials.append(line.split(':', 1))
+
+        self.username = base64.b64encode(credentials[0][0])
+        self.password = base64.b64encode(credentials[0][1])
+        self.dbName = credentials[1][1]
+        self.tableName = credentials[2][1]
+        self.hostName = base64.b64encode(credentials[3][1])
+        self.establishDbConnection()
+
+    def __del__(self):
+        # Close connection
+        self.connection.close()
+
+    def addValue(self, my_url, my_metadata):
+        # Create Table Object
```

Comment:
Add a `try..catch` block to make sure that connection exists before you perform any operations. In case of no connection, throw custom error to caller.